src/Altinn.App.Core/Internal/Patch/PatchService.cs

```diff
@@ -0,0 +1,136 @@
+        var modelType = _appModel.GetModelType(dataType.AppLogic.ClassRef);
+        var oldModel =
+            await _dataClient.GetFormData(instanceIdentifier.InstanceGuid, modelType, appIdentifier.Org, appIdentifier.App, instanceIdentifier.InstanceOwnerPartyId, Guid.Parse(dataElement.Id));
+        var oldModelNode = JsonSerializer.SerializeToNode(oldModel);
+            var patchResult = jsonPatch.Apply(oldModelNode);
+            if (!patchResult.IsSuccess)
+            {
+                bool testOperationFailed = patchResult.Error!.Contains("is not equal to the indicated value.");
+                return new DataPatchError()
+                {
+                    Title = testOperationFailed ? "Precondition in patch failed" : "Patch Operation Failed",
+                    Detail = patchResult.Error,
+                    ErrorType = testOperationFailed
+                        ? DataPatchErrorType.PatchTestFailed
+                        : DataPatchErrorType.DeserializationFailed,
+                    Extensions = new Dictionary<string, object?>()
+                    {
+                        { "previousModel", oldModel },
+                        { "patchOperationIndex", patchResult.Operation },
+                    }
+                };
+            }
+
+            var (model, error) = DeserializeModel(oldModel.GetType(), patchResult.Result!);
+            if (error is not null)
+            {
+                return new DataPatchError()
+                {
+                    Title = "Patch operation did not deserialize",
+                    Detail = error,
+                    ErrorType = DataPatchErrorType.DeserializationFailed
+                };
+            }
+            Guid dataElementId = Guid.Parse(dataElement.Id);
+            foreach (var dataProcessor in _dataProcessors)
+            {
+                await dataProcessor.ProcessDataWrite(instance, dataElementId, model, oldModel, language);
+            }
+
+            // Ensure that all lists are changed from null to empty list.
+            ObjectUtils.InitializeListsAndNullEmptyStrings(model);
+
+            var validationIssues = await _validationService.ValidateFormData(instance, dataElement, dataType, model, oldModel, ignoredValidators, language);
+
+            // Save Formdata to database
+            await _dataClient.UpdateData(
+                model,
+                instanceIdentifier.InstanceGuid,
+                modelType,
+                appIdentifier.Org,
+                appIdentifier.App,
+                instanceIdentifier.InstanceOwnerPartyId,
+                dataElementId);
+
+            return new DataPatchResult
+            {
+                NewDataModel = model,
+                ValidationIssues = validationIssues
+            };
+    }
+    
+    private static (object Model, string? Error) DeserializeModel(Type type, JsonNode patchResult)
```

Comment:
Her er vel et artig sted å bruke den nye `ServiceResult` for å se om den fungerer slik vi ønsker det?